haystack/nodes/prompt/invocation_layer/open_ai.py

```diff
@@ -150,23 +148,59 @@ def invoke(self, *args, **kwargs):
             "frequency_penalty": kwargs_with_defaults.get("frequency_penalty", 0),
             "logit_bias": kwargs_with_defaults.get("logit_bias", {}),
         }
+
+        return (prompt, base_payload, kwargs_with_defaults, stream, moderation)
+
+    def invoke(self, *args, **kwargs):
+        """
+        Invokes a prompt on the model. Based on the model, it takes in a prompt (or either a prompt or a list of messages)
+        and returns a list of responses using a REST invocation.
+
+        :return: The responses are being returned.
+
+        Note: Only kwargs relevant to OpenAI are passed to OpenAI rest API. Others kwargs are ignored.
+        For more details, see OpenAI [documentation](https://platform.openai.com/docs/api-reference/completions/create).
+        """
+        prompt, base_payload, kwargs_with_defaults, stream, moderation = self._prepare_invoke(*args, **kwargs)
+
         if moderation and check_openai_policy_violation(input=prompt, headers=self.headers):
             logger.info("Prompt '%s' will not be sent to OpenAI due to potential policy violation.", prompt)
             return []
-        responses = self._execute_openai_request(
-            prompt=prompt, base_payload=base_payload, kwargs_with_defaults=kwargs_with_defaults, stream=stream
-        )
```

Comment:
`ChatGPTInvocationLayer` inherits from `OpenAIInvocationLayer`
and has its own custom method [`_execute_openai_request`](https://github.com/deepset-ai/haystack/blob/d5d3a9eef4d4e66445e37b65bec3578a101d3075/haystack/nodes/prompt/invocation_layer/chatgpt.py#L46) that also creates requests with `messages` field (required in models such as `gpt3.5-turbo`).

Now this method is not invoked anymore, so we have errors like [this](https://github.com/deepset-ai/haystack/actions/runs/6415666359/job/17417960613).